subprojects/core-api/src/main/java/org/gradle/api/provider/Property.java

```diff
@@ -49,7 +49,7 @@
  * @since 4.3
  */
 @SupportsKotlinAssignmentOverloading
-public interface Property<T> extends Provider<T>, HasConfigurableValue {
+public interface Property<T> extends Provider<T>, HasConfigurableValue, SupportsConvention {
```

Comment:
:nail_care: This interface probably also needs `Property<T> unset()`, etc. overrides with the refined return type.